pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -0,0 +1,723 @@
+		cluster, err = e.setUnknown(cluster, apimgmtv3.ClusterConditionProvisioned, "")
+		if err != nil {
+			return cluster, err
+		}
+
+		if cluster.Status.AKSStatus.UpstreamSpec == nil {
+			cluster, err = e.setInitialUpstreamSpec(cluster)
+			if err != nil {
+				return cluster, err
+			}
+			return cluster, nil
+		}
+
+		e.clusterEnqueueAfter(cluster.Name, enqueueTime)
+		if failureMessage == "" {
+			logrus.Infof("waiting for cluster AKS [%s] to finish creating", cluster.Name)
+			return e.setUnknown(cluster, apimgmtv3.ClusterConditionProvisioned, "")
+		}
+		logrus.Infof("waiting for cluster AKS [%s] create failure to be resolved", cluster.Name)
+		return e.setFalse(cluster, apimgmtv3.ClusterConditionProvisioned, failureMessage)
+	case "active":
+		if cluster.Status.AKSStatus.UpstreamSpec == nil {
+			cluster, err = e.setInitialUpstreamSpec(cluster)
+			if err != nil {
+				return cluster, err
+			}
+			return cluster, nil
+		}
+
+		if apimgmtv3.ClusterConditionPending.IsUnknown(cluster) {
+			cluster = cluster.DeepCopy()
+			apimgmtv3.ClusterConditionPending.True(cluster)
+			cluster, err = e.clusterClient.Update(cluster)
+			if err != nil {
+				return cluster, err
+			}
+		}
+
+		if cluster.Spec.AKSConfig.Imported {
+			if cluster.Status.AKSStatus.UpstreamSpec == nil {
+				// non imported clusters will have already had upstream spec set
+				return e.setInitialUpstreamSpec(cluster)
+			}
+
+			if apimgmtv3.ClusterConditionPending.IsUnknown(cluster) {
+				cluster = cluster.DeepCopy()
+				apimgmtv3.ClusterConditionPending.True(cluster)
+				cluster, err = e.clusterClient.Update(cluster)
+				if err != nil {
+					return cluster, err
+				}
+			}
+		}
+
+		addNgMessage := "Cannot deploy agent without NodePools. Add a NodePool."
+		noNodePoolsOnSpec := len(cluster.Spec.AKSConfig.NodePools) == 0
+		noNodePoolsOnUpstreamSpec := len(cluster.Status.AKSStatus.UpstreamSpec.NodePools) == 0
+		if (cluster.Spec.AKSConfig.NodePools != nil && noNodePoolsOnSpec) || (cluster.Spec.AKSConfig.NodePools == nil && noNodePoolsOnUpstreamSpec) {
+			cluster, err = e.setFalse(cluster, apimgmtv3.ClusterConditionWaiting, addNgMessage)
+			if err != nil {
+				return cluster, err
+			}
+		} else {
+			if apimgmtv3.ClusterConditionWaiting.GetMessage(cluster) == addNgMessage {
+				cluster = cluster.DeepCopy()
+				apimgmtv3.ClusterConditionWaiting.Message(cluster, "Waiting for API to be available")
+				cluster, err = e.clusterClient.Update(cluster)
+				if err != nil {
+					return cluster, err
+				}
+			}
+		}
+
+		cluster, err = e.setTrue(cluster, apimgmtv3.ClusterConditionProvisioned, "")
+		if err != nil {
+			return cluster, err
+		}
+
+		// If there are no subnets it can be assumed that networking fields are not provided. In which case they
+		// should be created by the aks-operator, and needs to be copied to the cluster object.
+		if len(cluster.Status.AKSStatus.Subnets) == 0 {
+			subnets, _ := status["subnets"].([]interface{})
+			if len(subnets) != 0 {
```

Comment:
Does the aks operator generate subnets and virtual network if it isn't provided?

Reply:
This question needs to be addressed.

Reply:
I removed it for now